e2e/browser/test-app/components/appContent/index.tsx

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import { fetch as crossFetch, Headers } from "@inrupt/universal-fetch";
+import { useState } from "react";
+
+const OPENID_CONFIG_FILE_URL = "https://login.inrupt.com/.well-known/openid-configuration"
+
+export default function Home() {
+  const [fetchedData, setFetchedData] = useState("");
+  const headers = new Headers();
+  headers.append("Content-Type", "application/json");
+
+  const fetchData = async (options: RequestInit) => {
+    crossFetch(
+      OPENID_CONFIG_FILE_URL,
+      options
+    )
+      .then((response: Response) => response.json())
+      .then((response: Response) =>
+        setFetchedData(JSON.stringify(response, null, 2))
+      );
+  };
+
+  return (
+    <div>
+      <h1>
+        <code>@inrupt/universal-fetch</code> test browser app
+      </h1>
+      <p>
+        Content-type: <code data-testid="content-type">{headers.get("Content-Type")}</code>
+      </p>
+      <button onClick={() => fetchData({ headers })}>Fetch content</button>
+      <pre data-testid="data">{fetchedData}</pre>
```

Comment:
Shouldn't the `data-testid` be set by React when rendering the `fetchedData` successfully? Otherwise, isn't there a possibility that in the test there is a race condition between the test runner and the fetch result? It the `data-testid` is only set when the value is ready to be read, the test will block until the selector matches or it times out, so there is no race condition where React could be slow to render and the test would read the value before it is set.